spec/eof.md

```diff
@@ -385,20 +385,20 @@ Annotated examples of EOF formatted containers demonstrating several key feature
 
 ```solidity
 {
-    /// Takes [index][salt][init_data] as input,
+    /// Takes [tx_initcode_hash][salt][init_data] as input,
     /// creates contract and returns the address or failure otherwise
 
     // init_data.length can be 0, but the first 2 words are mandatory
     let size := calldatasize()
     if lt(size, 64) { revert(0, 0) }
 
-    let tx_initcode_index := calldataload(0)
+    let tx_initcode_hash := calldataload(0)
     let salt := calldataload(32)
```

Comment:
I think we should keccak the tx_initcode_hash with the salt.  Having salt only deployment as the principal toehold feels like opening up to a race, with no connection to the deployed code.  Subsequent ERC contracts can be deployed that do keep this feature, but if the salt only contract goes first the ERC contract set could get front run by literally anyone.

Reply:
yup, same realization in [the "big thread"](https://github.com/ipsilon/eof/issues/162). initcode_hash missing would actually not only open a race but break the feature of cross-chain deployments to same address, with no option of recovery! Will fix in a second.

is the addition of `0xef0001` magic just for good measure or do you have a scenario where a collision can be crafted somehow?

Reply:
I'm not a huge fan of the particular choice of magic value coinciding with EOF magic value. For now I'll leave it out for brevity, but I'll add a ticket to track figuring this out properly when time comes

Reply:
It can be another magic value, but I would like the address mining to be in it's own "namespace" and adding a fixed unique value does that. Any mini-salt is fine.

Reply:
ah, good point. How do we pick such "namespace" though... `0xff` again (and rely on different sizes of pre-image). Or do we want to be creative? I'll put in 0xff to publish and let's work from there

Reply:
ah, good point. How do we pick such "namespace" though... `0xff` again (and rely on different sizes of pre-image). Or do we want to be creative? I'll put in 0xff to publish and let's work from there

Reply:
My idea with `0xef0001` is that is is the premable to EOF.

For the ERCs I plan on `0x87650A`, `0x87650B`, and `0x87650C` assuming 8765 is the ERC number.

Reply:
> Having salt only deployment as the principal toehold feels like opening up to a race, with no connection to the deployed code.

The same can be said about init_data, right? Seems like the salt should be the hash of all of calldata for this to be secure.

Reply:
If the contract is being called via EXTDELEGATECALL, the caller address is also hashed into deploy address automatically, doesn't this alleviate front-running risk?

Reply:
Yes, is that the expected usage? EOAs are not able to use EXTDELEGATECALL (they can with EIP-7702 but we can't assume that of every EOA). I wouldn't make this contract insecure if used via normal EXTCALL.

Reply:
In the other thread we discussed the removal of magic value in final salt altogether. Also the init_data is now included in the init salt. Lastly, we're not neither restricting or requiring a particular way (delegate or not) of calling into here as discussed I think on last EOF call.

Many thanks for the feedback and discussion, I think we can resolve this conversation? I will do so unless anyone wants to add anything or we missed sth in the resulting spec in this branch.

Reply:
well, shoot. There might be a reason to keep a salt, and actually more unique than `0xff`, just like Danno suggests. If there are more factories, like for example one which doesn't include `init_data`, but `msg.sender` instead, that other factory would then need to include a magic value, to avoid conflict with the deployer contract (of course in a setting when EXTDELEGATECALL is used).

In a nutshell: is our Creator Contract distinguished enough to not come with a magic value in the `final_salt` calculation?

My gut tells me - yes, let the "derivative" factories worry about the magic value and avoiding unwanted salt-conflicts. WDYT?

Reply:
I think it is distinguished enough